website/docs/opengl/c1/2-hello-quad.md

```diff
@@ -0,0 +1,524 @@
+
+So, this explains why clearing the window works. You set the clear color, and it remains *as* the clear color until you change it.
+
+Now that we've got that, let's move on!
+
+## Vertex Array Objects (VAOs)
+A unique feature that modern OpenGL has is what is known as a Vertex Array Object, or VAO. It stores the state of various parameters, such as which buffers you wish to use, and the shader inputs.
+
+This is a required feature of modern OpenGL. You **must** have a VAO bound, otherwise your application won't work.
+
+Let's create the VAO!
+
+At the top of your class, add the following:
+
+```cs
+private static uint _vao;
+```
+
+Then, in your `OnLoad` method, add:
+
+```cs
+_vao = _gl.GenVertexArray();
+_gl.BindVertexArray(_vao);
+```
+
+What's going on here? First, we generate the VAO. Then, before we can use, or update it, we need to **bind** it.
+
+Binding is a term you will hear a lot with OpenGL, and in these tutorials too. Binding essentially means updating the current OpenGL state with the given object. So, in this example, when we bind the VAO, we update OpenGL's state so that the VAO we are currently using is now the VAO we have just created. If you don't bind an object before updating/using it, you will use or update the previously bound object instead, which can cause problems and headaches later.
+
+We're now ready to give the VAO some data. But first, we need a **vertex buffer**.
+
+## Vertex Buffer Objects (VBOs)
+Here is the part where we actually give the GPU some data it can work with. For 95% of graphics applications, a vertex buffer is **required** to display anything on screen.
+
+### What is a vertex buffer?
+Before we can continue, it's handy to know what a vertex buffer is, and what it does.
+
+Let's define both vertex, and buffer.
+
+* Vertex - A point where two lines, or edges meet.
+* Buffer - A region in memory that can be accessed to be written to, or read from.
+
+Knowing these definitions should give us a good idea of what a vertex buffer is. It's a region of memory, on the GPU, that stores points, which will get **rasterized** on-screen (usually in the form of triangles).
+
+### Vertex data
+Let's create the data that we will fill the vertex buffer with!
+
+In your `OnLoad` function, just after you call `_gl.BindVertexArray()`, add the following:
+
+```cs
+float[] vertices =
+{
+     0.5f,  0.5f, 0.0f,
+     0.5f, -0.5f, 0.0f,
+    -0.5f, -0.5f, 0.0f,
+    -0.5f,  0.5f, 0.0f
+};
+```
+
+### What makes up a quad?
+In modern graphics programming, you are expected to use triangles. While you may see options for quads, or lines, these are mostly deprecated.
```

Comment:
```suggestion
In modern graphics programming, you are expected to use triangles, lines, or points. While you may see options for quads, these are obsolete.
```

In [the knronos docs for primitives](https://www.khronos.org/opengl/wiki/Primitive), quads are deprecated as OpenGL 3.0, and have been removed from 3.1 and above. Lines and points are still part of the core specification.